framework/src/main/java/org/checkerframework/framework/stub/StubParser.java

```diff
@@ -1813,14 +1889,18 @@ private static void putOrAddToMap(
      * Just like Map.put, but merges with any existing value for the given key, instead ef replacing
      * it.
      */
-    private static void putNew(
+    private void putNew(
             Map<Element, AnnotatedTypeMirror> m, Element key, AnnotatedTypeMirror value) {
         if (key == null) {
             throw new BugInCF("StubParser: key is null");
         }
         if (m.containsKey(key)) {
             AnnotatedTypeMirror value2 = m.get(key);
-            AnnotatedTypeMerger.merge(value, value2);
+            if (isJdkAsStub) {
+                AnnotatedTypeMerger.merge(value2, value);
```

Comment:
This is another place where a comment would make it easier to understand why the order matters here.